### ISSUE TITLE
Remove environment from config file, just have one config per file

### ISSUE THREAD

**Issue by etiennedi** [creativesoftwarefdn/weaviate#800]
Currently we are grouping multiple sets of independent configs into a single config file. This is a 12-factor anti-pattern:

> In a twelve-factor app, env vars are granular controls, each fully orthogonal to other env vars. They are never grouped together as “environments”, but instead are independently managed for each deploy. This is a model that scales up smoothly as the app naturally expands into more deploys over its lifetime.

Source: https://12factor.net/config

More importantly though, we have seen exactly the issues that the 12-factor principles predict. For example, when switching between different setups (dev, runtime dc, etc.) we have mounted incorrect configs, that all of a sudden had a network configured.

So in essence we should go from

```yaml
environments:
   - name: one-environment
     optionA: bar
     optionB: baz

   - name: a-completely-different-environment
     optionA: bar
     optionB: baz
```

to

```yaml
optionA: foo
optionB: bar
```

**Comment by bobvanluijt** [comment]
Also, there might be deprecated things in the config files that can be removed.

**Comment by etiennedi** [comment]
Good point. In the MVP timeline we have a 2-3 or three days in the end to verify everything works as intended and is as polished as it should be. One of the things we can do there would be "clean up config management" in General. I.e. just timebox the cleanup and improvement of the config UX as much as possible.

**Comment by etiennedi** [comment]
We should also look into whether we could use https://github.com/spf13/viper Viper is 12-factor compatible by default and would provide a few benefits that could help us greatly at the moment:
For example we could set a relatively generic `config.yaml` and then overwrite individual values using environment variables. This way we don't need dozens of configuration files for each edge case. Especially with more plugins that we're currently adding, we could end up with many combinations that each require their own environment or file.

For example as part of #801 we'll add an option to spin up weaviate in docker-compose with keycloak running as an Identiy provider (for dev mode). With the current setup we'd need a whole new enviroment for this, with viper we could simply set something like `AUTHENTICATION_OIDC_ENABLED=true` and `AUTHENTICATION_OIDC_ISSUER:http://keycloak:8080`.

**Comment by etiennedi** [comment]
closing as split has been implemented as part of #826, outscoped viper to #848
